### ISSUE TITLE
Expose block transforms in the sidebar

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#27843]
Block transformations are a fundamental and powerful tool. They allow basic operations, like turning a paragraph into a heading, but also other important flows like turning a gallery into a slideshow, and generally being able to swap one block with another. In multi-selection contexts, it allows converting a series of paragraphs into a list, a series of images into a gallery, and many more.

It'd be good to explore ways in which we can increase awareness of these powerful transformations. One way would be to introduce transforms also on the sidebar, where the current block icon and description sit.

**Comment by mtias** [comment]
Thinking this would need to account for between transforms to other blocks, variations, and styles. cc @shaunandrews

**Comment by simison** [comment]
There sometimes are transformations between plugin blocks via core blocks (e.g. Jetpack tiled gallery transforms to core block, core block transforms to CoBlocks Masonry). Should those "two step" transforms be exposed in all places where we have block transforms? It could open up the number of transforms quite a bit.

**Comment by mtias** [comment]
Not sure I follow, do you mean to show "CoBlocks Masonry" as an option for "Jetpack Tiled" gallery by default given they share transformation to / from "Gallery"?

**Comment by jameskoster** [comment]
Certain block transforms have been exposed in the Inspector for some time, e.g. Group transforms:

<img width="280" height="186" alt="Image" src="https://github.com/user-attachments/assets/2733f5bb-0f26-420a-8682-4e4e2e207e47" />

Should we close this in favor of dedicated per-block issues for anything else we want to add?

**Comment by jasmussen** [comment]
Yes, let's. The act of closing here is also some fresh energy from folks subscribed, that would let us reopen if need be.
